### ISSUE TITLE
ng2-bootstrap pagination setPage need setTimeout

### ISSUE THREAD

**Issue by MichalObi** [valor-software/ngx-bootstrap#406]
What i'm trying to do is to set `this.currentPage` to 1 when Pagination events `numPages` is fired.

To make this work i need setTimeout inside numPages .

In **component.html** i have:

`<pagination  [totalItems]="totalItems" [itemsPerPage]="itemsPerPage" [(ngModel)]="currentPage" (pageChanged)="pageChanged($event)" (numPages)="numPages($event)" previousText="&lsaquo;" nextText="&rsaquo;"></pagination>`

In **component.ts** i have:

```
public setPage(pageNo:number):void {
    this.currentPage = pageNo;
};

public numPages($event:number):void {

   // Without setTimeout this doesn't fire.

    setTimeout(()=> {
        this.setPage(1);
    }, 50);

};
```

Whan i change the limit of items i can see that numPages `$event` is triggered immediately, but why i need to wait at least 50ms to make `setPage` work ?

**Comment by valorkin** [comment]
It was a hack some day to avoid dehydration issue in dev mode
it should be removed

**Comment by vemuez** [comment]
I had to do the same thing with pagination to make it go to page 1. Whenever the numPages is less than what it was before, it automatically goes to the latest page of the current totalResults. However, i wanted it to go to first page. Only fix currently was to setTimeout.

**Comment by YussufElarif** [comment]
I had to do the same thing with pagination in terms of setting a timeout to make it go to page 1. The only solution was to setTimeout because it would seem that every time I set the page to 1, the pagination component would overwrite the model and change it to the last page.

Check this plunker:
https://plnkr.co/edit/LDwF0xAbht4FohlPCVbZ

If plunkr has problems retrieving packages, the reproduced bug can be found here: https://github.com/vemuez/ngx-bootstrap-pagination

1. select "item" in drop down list
2. select page 3
3. select "brand" in drop down list (changes to page 1 as expected)

Now, follow these steps:
1. select "item" in drop down list
2. select the LAST page
3. select "brand" in drop down list  (doesn't change to page 1)

It overwrites the current page from page 1 to whatever page is the last page.
If the new list is smaller to an extent that it would change the number of pages to a number lower than the one previously, then it wouldn't work. 

e.g. Change the brand total to 95, it would still work but if you change it to 90, e.g. only 9 pages, it wont go to page 1
